plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -1292,6 +1302,85 @@ public void dropColumn(ConnectorSession session, ConnectorTableHandle tableHandl
         metastore.dropColumn(hiveTableHandle.getSchemaName(), hiveTableHandle.getTableName(), columnHandle.getName());
     }
 
+    @Override
+    public void setColumnType(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle columnHandle, Type type)
+    {
+        HiveTableHandle table = (HiveTableHandle) tableHandle;
+        failIfAvroSchemaIsSet(table);
+        HiveColumnHandle column = (HiveColumnHandle) columnHandle;
+
+        if (table.getPartitionNames().isPresent() && table.getPartitionNames().get().contains(column.getName())) {
+            throw new TrinoException(NOT_SUPPORTED, "Changing partition column types is not supported");
+        }
+        if (!canChangeColumnType(column.getType(), type)) {
+            throw new TrinoException(NOT_SUPPORTED, "Cannot change type from %s to %s".formatted(column.getType(), type));
+        }
+        metastore.setColumnType(table.getSchemaName(), table.getTableName(), column.getName(), toHiveType(type));
+    }
+
+    private static boolean canChangeColumnType(Type sourceType, Type newType)
```

Comment:
newType -> targetType (OR, sourceType -> oldType)

same to enhanced instanceof below